src/lib/components/Card/Card.tsx

```diff
@@ -22,29 +23,40 @@ export interface FlowbiteCardImageTheme {
   horizontal: FlowbiteBoolean;
 }
 
-export interface CardProps extends PropsWithChildren<ComponentProps<'div'>> {
+type CommonCardProps = PropsWithChildren<ComponentProps<'div'>> & {
   horizontal?: boolean;
   href?: string;
-  imgAlt?: string;
-  imgSrc?: string;
   theme?: DeepPartial<FlowbiteCardTheme>;
-}
+};
+
+export type CardProps =
+  | (
+      | { imgAlt?: string; imgSrc?: string; renderImage?: never }
+      | {
+          renderImage?: () => JSX.Element;
+          imgAlt?: never;
+          imgSrc?: never;
+        }
+    ) &
+      CommonCardProps;
 
-export const Card: FC<CardProps> = ({
-  children,
-  className,
-  horizontal,
-  href,
-  imgAlt,
-  imgSrc,
-  theme: customTheme = {},
-  ...props
-}) => {
+export const Card: FC<CardProps> = ({ children, className, horizontal, href, theme: customTheme = {}, ...props }) => {
   const Component = typeof href === 'undefined' ? 'div' : 'a';
-  const theirProps = props as object;
+  const theirProps = omit(props, ['renderImage', 'imgSrc', 'imgAlt']) as object;
```

Comment:
I find the gather above somewhat dangerous to begin with. I clean the props up here a bit by at least removing, what I do not want to have in the `Component` for sure.

Reply:
This is outdate, so, I'll mark it as resolved.